gate/tests/unit/system_token_test.go

```diff
@@ -0,0 +1,53 @@
+package unit
+
+import (
+	"github.com/ZupIT/charlescd/gate/internal/domain"
+	"github.com/ZupIT/charlescd/gate/internal/logging"
+	systemTokenInteractor "github.com/ZupIT/charlescd/gate/internal/use_case/system_token"
+	mocks "github.com/ZupIT/charlescd/gate/tests/unit/mocks/repository"
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"testing"
+)
+
+type SystemTokenSuite struct {
+	suite.Suite
+	getSystemToken systemTokenInteractor.GetSystemToken
+	systemTokenRepository      *mocks.SystemTokenRepository
+}
+
+func (st *SystemTokenSuite) SetupSuite() {
+	st.systemTokenRepository = new(mocks.SystemTokenRepository)
+	st.getSystemToken = systemTokenInteractor.NewGetSystemToken(st.systemTokenRepository)
+}
+
+func (st *SystemTokenSuite) SetupTest() {
+	st.SetupSuite()
+}
+
+func TestSuite(t *testing.T) {
+	suite.Run(t, new(SystemTokenSuite))
+}
+
+func (st *SystemTokenSuite) TestGetByID() {
+	st.systemTokenRepository.On("FindById", mock.Anything).Return(domain.SystemToken{ID: uuid.New()}, nil)
```

Comment:
Suggestion:

To make sure that the correct input is going to be received, you could initialize a system token and reference it here:
```suggestion
	st.systemTokenRepository.On("FindById", systemToken.ID).Return(systemToken, nil)
```